code/DEMO_ProcessRCS.m

```diff
@@ -0,0 +1,261 @@
+% TimeDomain data
+disp('Checking for Time Domain Data')
+TD_fileToLoad = [folderPath filesep 'RawDataTD.json'];
+if isfile(TD_fileToLoad)
+    jsonobj_TD = deserializeJSON(TD_fileToLoad);
+    if ~isempty(jsonobj_TD.TimeDomainData)
+        disp('Loading Time Domain Data')
+        [outtable_TD, srates_TD] = createTimeDomainTable(jsonobj_TD);
+        disp('Creating derivedTimes for time domain:')
+        timeDomainData = assignTime(outtable_TD);
+    else
+        timeDomainData = [];
+    end
+else
+    timeDomainData = [];
+end
+
+%%
+% Accelerometer data
+disp('Checking for Accelerometer Data')
+Accel_fileToLoad = [folderPath filesep 'RawDataAccel.json'];
+if isfile(Accel_fileToLoad)
+    jsonobj_Accel = deserializeJSON(Accel_fileToLoad);
+    if ~isempty(jsonobj_Accel.AccelData)
+        disp('Loading Accelerometer Data')
+        [outtable_Accel, srates_Accel] = createAccelTable(jsonobj_Accel);
+        disp('Creating derivedTimes for accelerometer:')
+        AccelData = assignTime(outtable_Accel);
+    else
+        AccelData = [];
+    end
+else
+    AccelData = [];
+end
+
+%%
+% Power data
+disp('Checking for Power Data')
+Power_fileToLoad = [folderPath filesep 'RawDataPower.json'];
+if isfile(Power_fileToLoad)
+    disp('Loading Power Data')
+    % Checking if power data is empty happens within createPowerTable
+    % function
+    [outtable_Power] = createPowerTable(folderPath);
+    
+    % Calculate power band cutoffs (in Hz) and add column to powerSettings
+    if ~isempty(outtable_Power)
+        % Translate powerSettings.powerBands into Hz
+        numSettings = size(powerSettings,1);
+        for iSetting = 1:numSettings
+            powerBands_toConvert = powerSettings.powerBands{iSetting};
+            currentTDsampleRate = powerSettings.TDsampleRates{iSetting};
+            currentFFTconfig = powerSettings.fftConfig{iSetting};
+            [currentPowerBands] = getPowerBands(powerBands_toConvert,currentFFTconfig,currentTDsampleRate);
+            powerSettings.powerBandsInHz(iSetting) = currentPowerBands;
+        end
+        
+        % Add samplerate and packetsizes column to outtable_Power -- samplerate is inverse
+        % of fftConfig.interval; in principle this interval could change
+        % over the course of the recording
+        
+        % Determine if more than one sampling rate across recording
+        for iSetting = 1:numSettings
+            all_powerFs(iSetting) =  1/((powerSettings.fftConfig{iSetting}.interval)/1000);
+        end
+        
+        if length(unique(all_powerFs)) > 1
+            error('More than one sampling rate for power channels -- code development needed')
+            % Need to loop through each setting in powerSettings; find
+            % closest times between powerSettings and outtable_Power to
+            % assign corresponding samplerate
+        else
+            % Same sample rate for power data for the full file
+            powerDomain_sampleRate = unique(all_powerFs);
```

Comment:
I know this is only an example, but I would change sampleRate to sampleResolution. If I understand the code above correctly that is.

Reply:
Here, `powerDomain_sampleRate` has been converted to Hz (e.g. if interval for fft was 100 [ms], then `powerDomain_sampleRate` will equal 10). The purpose of this is to get these data into a similar format as other data streams, so `assignTime` can be used. Do you still think the name should be changed?

Reply:
No, this is fine.